### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by oleg-rdk**

### CODE REVIEW

File: .meteor/dev_bundle [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/oleg/.meteor/packages/meteor-tool/.1.3.4_4.a3kdwt++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle
```
Comment (oleg-rdk) [inline .meteor/dev_bundle]: I will add to .gitignore

File: client/modules/dashboard/components/data_visualisation.jsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+import Toggle from 'material-ui/Toggle';
+import IconButton from 'material-ui/IconButton';
+import TextField from 'material-ui/TextField';
+
+import ChartCanvas from '../../workplace/containers/chart_canvas';
+
+import '/node_modules/react-grid-layout/css/styles.css';
+import '/node_modules/react-resizable/css/styles.css';
+
+const styles = {
```
Comment (oleg-rdk) [inline client/modules/dashboard/components/data_visualisation.jsx]: I will replace to separate import.style file

File: client/modules/dashboard/actions/dashboard.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  routeTo({ FlowRouter }, { name, params, queryParams }) {
+    const path = FlowRouter.path(name, params, queryParams);
+    FlowRouter.go(path);
+  },
+  renameDashboard({ Meteor }, { dashboardId, newName }) {
+    Meteor.call('dashboard.rename', { dashboardId, newName }, (err) => {
+      if (err) throw new Meteor.Error(err);
+    });
+  },
+};
```
Comment (Serhiy91) [inline client/modules/dashboard/actions/dashboard.js]: rewrite these funcs to 
callMeteorMethod(methodName, params) {
    Meteor.call(methodName, params, (err) => {
        if (err) throw new Meteor.Error(err);
     });
}

File: client/modules/dashboard/components/dashboard.jsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
+import React, { PropTypes } from 'react';
```
Comment (Serhiy91) [inline client/modules/dashboard/components/dashboard.jsx]: first without {} than {}

File: client/modules/dashboard/components/dashboard.jsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    return _.isEqual(makeObject(newLocations), makeObject(previousLocations));
+  }
+  saveChartsLocation(newLocations, layout) {
+    const dashboardId = this.props.dashboard._id;
+    const breakpoint = _.keys(layout)[0];
+    const previousLocations = this.getChartsLocation()[breakpoint];
+    if (!this.isEqualLocations({ newLocations, previousLocations })) {
+      this.props.saveChartsLocation({ dashboardId, newLocations, breakpoint });
+    }
+  }
```
Comment (Serhiy91) [inline client/modules/dashboard/components/dashboard.jsx]: delete handleAddChartButton

File: client/modules/dashboard/components/handleAddChart.jsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from 'react';
+
+import i18n from 'meteor/universe:i18n';
+
+import IconMenu from 'material-ui/IconMenu';
+import MenuItem from 'material-ui/MenuItem';
+import IconButton from 'material-ui/IconButton';
+import FloatingActionButton from 'material-ui/FloatingActionButton';
+import ContentAdd from 'material-ui/svg-icons/content/add';
+
```
Comment (Serhiy91) [inline client/modules/dashboard/components/handleAddChart.jsx]: tow components with Dashboard name

File: client/modules/dashboard/components/data_visualisation.jsx [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+            </Table>
+            :
+            <ChartCanvas
+              data={data}
+              fields={queryObject.fields}
+              chartType={chartType}
+              heightIsFixed={false}
+              needRedraw={needRedraw}
+            />
+          }
```
Comment (trsdln) [inline client/modules/dashboard/components/data_visualisation.jsx]: it's completely unreadable

File: client/modules/workplace/components/workplace.import.styl [hunk 8]
```diff
@@ -65,16 +65,27 @@ $area-key()
...
-    margin-right 15px
+    right 15px
+    bottom 0px
+    .text-pagination
+      padding 16px
+      font-size 14px
+    .action-pagination,
+    .text-pagination
+      float right
+      margin-right 15px
```
Comment (trsdln) [inline client/modules/workplace/components/workplace.import.styl]: What about mixins & variables?

File: .meteor/dev_bundle [hunk 9]
```diff
@@ -1 +1 @@
-/Users/zinchuk/.meteor/packages/meteor-tool/.1.3.4_4.1ku8cvh++os.osx.x86_64+web.browser+web.cordova/mt-os.osx.x86_64/dev_bundle
\ No newline at end of file
+/home/oleg/.meteor/packages/meteor-tool/.1.3.4_4.a3kdwt++os.linux.x86_64+web.browser+web.cordova/mt-os.linux.x86_64/dev_bundle
```
Comment (trsdln) [inline .meteor/dev_bundle]: @oleg-rdk it shouldn't be here.

File: config/settings.json [hunk 10]
```diff
@@ -3,7 +3,7 @@
     "quasar": {
       "host": "localhost",
       "port": "20223",
-      "db": "tutorial"
+      "db": "databasel"
```
Comment (trsdln) [inline config/settings.json]: probably config folder should be outside of application folder in order to avoid unintentional publishing of sensitive data.

**Comment by trsdln** [comment]
I'm not react expert, but components I saw in this PR look completely unreadable

File: client/modules/workplace/containers/workplace.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+import { useDeps, composeWithTracker, composeAll } from 'mantra-core';
+import Workplace from '../components/workplace.jsx';
+
```
Comment (Serhiy91) [inline client/modules/workplace/containers/workplace.js]: implement func for reset chart state

File: client/modules/workplace/components/preview/preview_handleDashboards.jsx [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+    super(props);
+    this.state = {
+      valueMultiple: this.props.parsedDashboards
+        .filter(board => board.added)
+        .map(board => board.id),
+    };
+    this.handleOpenMenu = this.handleOpenMenu.bind(this);
+    this.handleOnRequestChange = this.handleOnRequestChange.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
```
Comment (Serhiy91) [inline client/modules/workplace/components/preview/preview_handleDashboards.jsx]: logic of handleChange replace into actions

File: client/modules/dashboard/components/data_visualisation.jsx (hunk 7 above)
Comment (oleg-rdk) [inline client/modules/dashboard/components/data_visualisation.jsx]: @trsdln Completely agree. We plan to refactor it as soon as possible

**Pull request merged**